practices/managing-static-assets.md

```diff
@@ -0,0 +1,16 @@
+# Managing static assets
+
+Static assets are files that don't need to be generated, modified or processed by an application before being delivered to the user. This typically includes CSS, JavaScript, images and web fonts.
+
+This content rarely changes and its contents don't depend on user input or preferences.
+
+For this kind of static resources it is our choice to **self-host them on our servers** instead of including them on our pages from an external CDN.
+
+## Benefits of self-hosting static assets
+
+* **Resilience.** As long as our site is up, the resource will be up. We don't depend on other servers or additional CDNs and we won't be affected by their outages, or closure of the service.
```

Comment:
Resilience - as I understand it - is:

> an outcome of a highly automated, well architected and well tested system, predicated on the following common characteristics:    
> 
> - **Adaptive** — the ability to adapt to changing conditions.
> - **Self Healing** — the ability to recover from failures autonomously.
> - **Predictable** — the ability to consistently define responses to failures. This is a prerequisite to support the automation necessary to be adaptive and self healing.

If we are suggesting some method of including static assets on our servers as _a fallback_ for when the CDN fails, then that would be considered a resilient approach.

If however we're proposing moving stuff from a CDN to our own servers, is that not making it more fragile? Do we have better uptime than our CDN?

Reply:
Ah hold on … do you mean "only host on Fastly", rather than "don't host on _any_ CDN"?

Reply:
> Ah hold on … do you mean "only host on Fastly", rather than "don't host on _any_ CDN"?

I mean "hosting jQuery ourselves" vs "loading jQuery from https://code.jquery.com/". What would be the appropriate word to describe that beyond "resilience"?